emcc.py

```diff
@@ -1438,7 +1438,9 @@ def default_setting(name, new_default):
     if name not in settings_map:
       setattr(settings, name, new_default)
 
-  if settings.OPT_LEVEL >= 1:
+  if settings.SAFE_HEAP >= 1:
+    default_setting('ASSERTIONS', 1)
+  elif settings.OPT_LEVEL >= 1:
     default_setting('ASSERTIONS', 0)
   if settings.SHRINK_LEVEL >= 2:
```

Comment:
```suggestion

  if settings.SHRINK_LEVEL >= 2:
```
(seems nice to add a line break here so visually it is clear the if-else chain is separate)

Reply:
> 

I think you are right... I'll revert this part and ad an error regarding `MINIMAL_RUNTIME`

Reply:
I found a better solution which is just to include the `asset` function when SAFE_HEAP is enabled.